db/hints/resource_manager.cc

```diff
@@ -59,8 +59,13 @@ future<semaphore_units<semaphore_default_exception_factory>> resource_manager::g
     }
     seastar::semaphore& limiter = *it->second;
 
+    // Take into account that hints may be sent in parallel by many nodes. In the worst case all other nodes may be
+    // sending hints to the same shard on the same node, so scale by live members count (excluding the destination).
+    const size_t live_members_count = _gossiper_ptr->get_live_members_count();
+    const size_t scale_factor = (live_members_count > 1) ? live_members_count - 1 : 1;
```

Comment:
`const size_t scale_factor = std::max(live_members_count - 1, 1);`

Would make the intent more apparent.

Reply:
I wrote it like this in order to avoid underflow in rare case when `live_members_count == 0` (`size_t` is unsigned). I think it could happen when all other nodes are not alive, and we are shutting down. I can add a comment to make it more clear.

Another option would be a compromise: `const size_t scale_factor = std::max(live_members_count, 2) - 1;`...

...or not to care about underflow at all (nothing should break, we will just limit sending to 1 hint per endpoint), but I'd rather avoid over/underflows if possible.

Reply:
I think we should avoid under/over flow, so either the current code or the `std::max()` version is fine.

Reply:
OK, I added a comment.